components/Markdown.tsx

```diff
@@ -163,14 +58,106 @@ function Markdown(props: MarkdownProps) {
   if (!props.source) {
     return null;
   }
+
+  marked.use({
+    renderer: ({
+      heading(text: string, level: number) {
+        const slug = slugify(text);
+        return `
+          <h${level}>
+            <a name="${slug}" class="anchor" href="#${slug}">
+              <span class="octicon-link"></span>
+            </a>
+            ${text}
+          </h${level}>`;
+      },
+      link(href, title, text) {
+        const url = href ? transformLinkUri(props.displayURL)(href) : "";
+        return `<a ${url ? `href="${url}"` : ""} ${
+          title ? `title="${title}"` : ""
+        }>${text}</a>`;
+      },
+      image(href, title, text) {
+        const url = href ? transformImageUri(props.sourceURL)(href) : "";
+        return `<img ${url ? `src="${url}"` : ""} ${
+          text ? `alt="${text}"` : ""
+        } ${title ? `title="${title}"` : ""} style="max-width:100%;">`;
+      },
+      html(html: string) {
+        const images: RegExpMatchArray[] = [...html.matchAll(/src="([^"]*)"/g)];
+        images.forEach((a) => {
+          const original = a[1];
+          const final = transformImageUri(props.sourceURL)(original);
+          console.log(original, final);
+          html = html.replace(`src="${original}"`, `src="${final}"`);
+        });
+        const links: RegExpMatchArray[] = [...html.matchAll(/href="([^"]*)"/g)];
+        links.forEach((a) => {
+          const original = a[1];
+          const final = transformLinkUri(props.displayURL)(original);
+          console.log(original, final);
+          html = html.replace(`href="${original}"`, `href="${final}"`);
+        });
+        return html;
+      },
+    } as Partial<Renderer>) as any,
+  });
+
+  const raw = marked(props.source, {
+    gfm: true,
+    headerIds: true,
+    sanitizer: dompurify.sanitize,
+    highlight: (code, language) =>
+      renderToStaticMarkup(
+        <RawCodeBlock
+          code={code}
+          language={language as any}
+          disablePrefixes={true}
+          enableLineRef={false}
+        />
+      ),
```

Comment:
This isn't great. I still need to ponder on how to do this without including multiple highlighters / themes. I will revisit this later.

Reply:
ok